src/MarshalDelegatorsConfig.php

```diff
@@ -42,9 +42,13 @@ private function delegateFactories(
         // Marshal from factory
         $serviceFactory = $dependencies->get($service);
         return static function (ContainerInterface $container) use ($service, $serviceFactory) {
-            return is_callable($serviceFactory)
-                ? $serviceFactory($container, $service)
-                : (new $serviceFactory())($container, $service);
+            if (is_callable($serviceFactory)) {
```

Comment:
Should we use the YODA condition syntax?

Any other codes on this repository use above syntax.

And it should be `if (true === is_callable($serviceFactory)) {`